app/services/potential_duplicate_finder_service.rb

```diff
@@ -15,10 +17,15 @@ def self.duplicates_finder
           .where(patient_program: { program_id: 33, voided: 0 })
           .select('person.person_id, person.birthdate, person.gender,
                 person_name.given_name, person_name.family_name,
-                person_name.middle_name,person_address.address2 AS home_district,
+                def self.fetch_patients
+                      Person.joins(:names, :addresses)
+                                 .joins(patient: :patient_programs)
+                                .where(patient_program: { program_id: 33, voided: 0 })
+                                .select('person.person_id, person.birthdate, person.gender, ...')
+                end,
                 person_address.neighborhood_cell AS home_village,
```

Comment:
from line 13 - 36 the code should be replaced by this 
```suggestion
                fetch_patients.each do |primary_patient|
                   potential_duplicates = []
      fuzzy_potential_duplicates = []
                  soundex_duplicates = []
                  soundex_potentials = []

      fetch_patients(use_batches: true).each do |batch|
        batch.each do |potential_duplicate|
```
on line 53 add another end to close the loop
```suggestion
                end
```
a helper method can be written like this 
```suggestion
                def self.fetch_patients(use_batches: false)
    patients_query = Person.joins(:names, :addresses)
                           .joins(patient: :patient_programs)
                           .where(patient_program: { program_id: 33, voided: 0 })
                           .select('person.person_id, person.birthdate, person.gender,
                                   person_name.given_name, person_name.family_name, 
                                   person_name.middle_name, person_address.address2 AS home_district,
                                   person_address.neighborhood_cell AS home_village, 
                                   person_address.county_district AS home_traditional_authority')

    if use_batches
      patients_query.in_batches(of: 1000)
    else
      patients_query.find_each
    end
  end
```